src/go/rpk/pkg/config/params.go

```diff
@@ -546,7 +546,14 @@ func (c *Config) addUnsetDefaults() {
 		brokers = []string{"127.0.0.1:9092"}
 	}
 
-	if len(r.AdminAPI.Addresses) == 0 {
+	addrs := r.AdminAPI.Addresses
```

Comment:
It should be straightforward to add an integration (ducktape) test for this: when redpanda is run in ducktape -- add a method to RpkRemote that requires admin API config, and check that calling out to it on a redpanda node works.

Reply:
unit tests have been added for this function itself 👍